web_time_range_menu_custom/static/src/js/domain.js

```diff
@@ -0,0 +1,236 @@
+/* Copyright 2021 Tecnativa - Alexandre D. Díaz
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl). */
+odoo.define("web_time_range_menu_custom.Domain", function(require) {
+    "use strict";
+
+    const Domain = require("web.Domain");
+
+    Domain.include({
+        /**
+         * @override
+         */
+        constructCustomDomain: function(
+            fieldName,
+            period,
+            type,
+            comparisonPeriod,
+            periodCustom,
+            comparisonPeriodCustom
+        ) {
+            let leftBoundaryParams = {};
+            let rightBoundaryParams = {};
+            let offsetPeriodParams = 0;
+            // Cloned function from web.domain
+            // Necessary because inner function usage
```

Comment:
Ouch, such a rigid implementation from Odoo. It's hard to see what's monkeypatched and what's the original code. Please document in the code.

Also put some red flags and TODOs to check this in migrations to be aligned with the upstream code.